src/SignalR/clients/ts/FunctionalTests/Startup.cs

```diff
@@ -73,17 +77,33 @@ public void ConfigureServices(IServiceCollection services)
                     {
                         OnMessageReceived = context =>
                         {
-                            var signalRTokenHeader = context.Request.Query["access_token"];
+                            var endpoint = context.HttpContext.Features.Get<IEndpointFeature>()?.Endpoint;
+                            if (endpoint != null && endpoint.Metadata.GetMetadata<HubMetadata>() != null)
```

Comment:
@davidfowl this particular change was suggested by @BrennanConroy. In lieu of checking if a particular request is a WebSocket request, this always checks if the query string is specified. However it needs to read the Auth header if available.

Reply:
I think this is reasonable.

The slight concern I have is that this could encourage people to put an access_token in the query string when they could have set an Authorization header. We'd like to encourage using the header over the query string since the query string is more likely to be logged.

I don't think that concern is big enough to warrant changing this logic though.